tests/api2/test_040_ad_user_group_cache.py

```diff
@@ -1,165 +1,127 @@
-    })
-    assert results.status_code == 200, results.text
-    assert len(results.json()) > 0, results.text
-    cache_names = [x['name'] for x in results.json()]
+    cache_names = set([x['name'] for x in call(
+       'group.query',
+       [['local', '=', False]],
+       {'extra': {'search_dscache': True}}
+    )])
 
-    for entry in wbinfo_entries:
-        assert entry in cache_names, str(cache_names)
+    assert wbinfo_entries == cache_names
 
 
-@pytest.mark.dependency(name="REBUILD_AD_CACHE")
-def test_09_check_directoryservices_cache_refresh(request):
+def test_check_directoryservices_cache_refresh(do_ad_connection):
     """
     This test validates that middleware can successfully rebuild the
     directory services cache from scratch using the public API.
 
     This currently happens once per 24 hours. Result of failure here will
     be lack of users/groups visible in webui.
     """
-    depends(request, ["AD_USERS_CACHED", "AD_GROUPS_CACHED"], scope="session")
-    rebuild_ok = False
 
-    """
-    Cache resides in tdb files. Remove the files to clear cache.
-    """
+    # Cache resides in tdb files. Remove the files to clear cache.
     cmd = 'rm -f /root/tdb/persistent/*'
     results = SSH_TEST(cmd, user, password)
     assert results['result'] is True, results['output']
 
-    """
-    directoryservices.cache_refresh job causes us to rebuild / refresh
-    LDAP / AD users.
-    """
-    results = GET('/directoryservices/cache_refresh/')
-    assert results.status_code == 200, results.text
-    if results.status_code == 200:
-        refresh_job = results.json()
-        job_status = wait_on_job(refresh_job, 180)
-        assert job_status['state'] == 'SUCCESS', str(job_status['results'])
-        if job_status['state'] == 'SUCCESS':
-            rebuild_ok = True
+    # directoryservices.cache_refresh job causes us to rebuild / refresh LDAP / AD users.
+    call('directoryservices.cache.refresh', job=True)
 
-    """
-    Verify that the AD user / group cache was rebuilt successfully.
-    """
-    if rebuild_ok:
-        results = GET('/group', payload={
-            'query-filters': [['local', '=', False]],
-            'query-options': {'extra': {"search_dscache": True}},
-        })
-        assert results.status_code == 200, results.text
-        assert len(results.json()) > 0, results.text
-
-        results = GET('/user', payload={
-            'query-filters': [['local', '=', False]],
-            'query-options': {'extra': {"search_dscache": True}},
-        })
-        assert results.status_code == 200, results.text
-        assert len(results.json()) > 0, results.text
-
-
-@pytest.mark.dependency(name="LAZY_INITIALIZATION_BY_NAME")
-def test_10_check_lazy_initialization_of_users_and_groups_by_name(request):
+    users = set([x['username'] for x in call(
+       'user.query',
+       [['local', '=', False]],
+       {'extra': {'search_dscache': True}}
+    )])
+
+    assert len(users) > 0
```

Comment:
Maybe also confirm actual unique names, not just that some names exist or do we have an expectation on the number of entries that we can test against?